### PR TITLE
checks: Ensure version match between releases.json and meson.build

### PR DESCRIPTION
**Pull request by thiblahute**
And fix all mistmatches

### CODE REVIEW

File: subprojects/packagefiles/abseil-cpp/meson.build [hunk 1]
```diff
@@ -1,5 +1,5 @@
 project('abseil-cpp', 'cpp',
-  version : '20210324.1',
+  version : '20210324.2',
```
Comment (eli-schwartz) [inline subprojects/packagefiles/abseil-cpp/meson.build:2]: Haha, after complaining about vorbis  forgetting to bump the version in meson.build, the *only* other wrap with the problem is... my fault? :D

Comment (thiblahute) [reply subprojects/packagefiles/abseil-cpp/meson.build:2]: :-)

File: tools/sanity_checks.py [hunk 2]
```diff
@@ -72,6 +72,21 @@ def test_releases_json(self):
...
                 self.assertIn(k, PERMITTED_KEYS)
 
+    def get_patch_path(self, wrap_section):
+        patch_directory = wrap_section.get('patch_directory')
+        if patch_directory:
+            return Path('subprojects', 'packagefiles', patch_directory)
+
+        return None
+
+    def check_meson_version(self, name: str, version: str,patch_path: str, builddir: str = '_build'):
```
Comment (eli-schwartz) [inline tools/sanity_checks.py:82]:
```suggestion
    def check_meson_version(self, name: str, version: str, patch_path: str, builddir: str = '_build'):
```

File: tools/sanity_checks.py [hunk 3]
```diff
@@ -72,6 +72,21 @@ def test_releases_json(self):
...
+        with self.subTest(step="check_meson_version"):
+            with open(Path(builddir) / "meson-info/intro-projectinfo.json") as project_info_file:
+                project_info = json.load(project_info_file)
+                subproject, = [subproj for subproj in project_info["subprojects"] if subproj["name"] == name]
+                if subproject['version'] != 'undefined' and not patch_path:
```
Comment (eli-schwartz) [inline tools/sanity_checks.py:87]: So we only run the check if the patch_path does not exist -- i.e. meson.build comes from upstream? Oops.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: tools/sanity_checks.py [hunk 4]
```diff
@@ -72,6 +72,21 @@ def test_releases_json(self):
...
+                    self.assertEqual(subproject['version'], version, f"for '{name}'")
```
[the remark this answers is not in the document]
Comment (thiblahute) [reply tools/sanity_checks.py]: Makes sense, removed it.

**Comment by eli-schwartz** [comment]
Thanks!

**Pull request merged**